pkg/csi/cinder/controllerserver.go

```diff
@@ -48,6 +51,13 @@ const (
 func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
 	klog.V(4).Infof("CreateVolume: called with args %+v", protosanitizer.StripSecrets(*req))
 
+	// Volume cloud
+	volCloud := req.GetSecrets()["cloud"]
```

Comment:
How does `GetSecrets()` work?

Reply:
CSI [spec](https://github.com/container-storage-interface/spec/blob/master/spec.md) handle passing secrets through RPC calls, so some calls have a `secrets` field implemented in request.

In this case [CreateVolume](https://github.com/container-storage-interface/spec/blob/master/spec.md#createvolume) request support passing some secret data:
![image](https://github.com/kubernetes/cloud-provider-openstack/assets/4145084/5be68615-0bb9-4aff-8fcd-7f4a230a4554)

How to use it:
- You have to create your k8s secret in namespace you want with your data. In our case secret contains only a key `cloud` with the "could name" as value nothing really sensitive but it is the only way I found to be able to propagate cloud hosting a volume.
- Then you have to refer this secret in your StorageClass object as documented [here](https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html) which permit to specify to external-provisioner to feed RPC call secrets with your kubernetes secret data.
- In this way CSI spec offert a method [GetSecrets](https://github.com/container-storage-interface/spec/blob/v1.8.0/lib/go/csi/csi.pb.go#L979) to retrieve this data from csi-plugin implmentation (in our case cinder-csi-plugin).

This is my third idea to resolve this issue (mention in issue description)

> I have 2 ideas to troubleshoot this issue:
> - query each managed OS clouds to find one who had a volume with searched ID (That's linear complexity)
> - implement a kubernetes client and as kubernetes API to find (bad idea, I just realized that volume_id is openstack id not pv name)
> - maybe use secrets field as it seems suggested in [external-provisioner discussion](https://github.com/kubernetes-csi/external-provisioner/issues/87)